docs/guide/input-validation.md

```diff
@@ -442,6 +442,381 @@ class EntryForm extends Model
+            foreach ($this->attributes as $attribute) {
+                $attributes[] = $attribute[0] === '!' ? substr($attribute, 1) : $attribute;
+            }
+        }
+
+        foreach ($attributes as $attribute) {
+            $skip = $this->skipOnError && $model->hasErrors($attribute)
+                || $this->skipOnEmpty && $this->isEmpty($model->$attribute);
+            if ($skip) {
+                // Skip validation if at least one attribute is empty or already has error
+                // (according skipOnError and skipOnEmpty options must be set to true
+                return;
+            }
+        }
+
+        if ($this->batch) {
+            // Validate all attributes at once
+            if ($this->when === null || call_user_func($this->when, $model, $attribute)) {
+                // Pass array with all attributes instead of one attribute
+                $this->validateAttribute($model, $attributes);
+            }
+        } else {
+            // Validate each attribute separately using the same validation logic
+            foreach ($attributes as $attribute) {
+                if ($this->when === null || call_user_func($this->when, $model, $attribute)) {
+                    $this->validateAttribute($model, $attribute);
+                }
+            }
+        }
+    }
+}
+```
+
+Then we need to create custom validator and use the created trait:
+
+```php
+<?php
+
+namespace app\components;
+
+use yii\validators\Validator;
+
+class CustomValidator extends Validator
+{
+    use BatchValidationTrait;
+}
+```
+
+To support inline validation as well we can extend default inline validator and also use this trait:
+
+```php
+<?php
+
+namespace app\components;
+
+use yii\validators\InlineValidator;
+
+class CustomInlineValidator extends InlineValidator
+{
+    use BatchValidationTrait;
+}
+```
+
+Couple more changes are needed.
+
+First to use our `CustomInlineValidator` instead of default `InlineValidator` we need to override
+[[yii\validators\Validator::createValidator()]] method in `CustomValidator`:
+
+```php
+public static function createValidator($type, $model, $attributes, $params = [])
+{
+    $params['attributes'] = $attributes;
+
+    if ($type instanceof \Closure || $model->hasMethod($type)) {
+        // method-based validator
+        // The following line is changed to use our CustomInlineValidator
+        $params['class'] = __NAMESPACE__ . '\CustomInlineValidator';
+        $params['method'] = $type;
+    } else {
+        if (isset(static::$builtInValidators[$type])) {
+            $type = static::$builtInValidators[$type];
+        }
+        if (is_array($type)) {
+            $params = array_merge($type, $params);
+        } else {
+            $params['class'] = $type;
+        }
+    }
+
+    return Yii::createObject($params);
+}
+```
+
+And finally to support our custom validator in model we can create the trait and override
+[[yii\base\Model::createValidators()]] like this:
```

Comment:
In other places of the docs (even in the same file) it's used without leading backslash. Used it as an example.